src/govuk/common.mjs

```diff
@@ -129,3 +129,51 @@ export function extractConfigByNamespace (configObject, namespace) {
   }
   return newObject
 }
+
+/**
+ * Normalise data
+ *
+ * 'If it looks like a duck, and it quacks like a duck…' 🦆
+ *
+ * If the passed value looks like a boolean or a number, convert it to a boolean
+ * or number.
+ *
+ * Designed to be used to convert config passed via data attributes (which are
+ * always strings) into something sensible.
+ *
+ * @param {String} value - The value to normalise
```

Comment:
I know this is generally called from `normaliseDataset()` but [DOMStringMap](https://developer.mozilla.org/en-US/docs/Web/API/DOMStringMap) value might be [`undefined`](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.domstringmap.html)

**Element _without_ `data-missing`**
```mjs
normaliseDataset($module.dataset.missing)
```

It'll throw an error when `value.trim()` is run:
```console
TypeError: undefined is not an object (evaluating 'undefined.trim')
```

```suggestion
 * @param {String|undefined} value - The value to normalise
```

Reply:
Maybe this function should be called `normaliseString`?

It's intended to be used by `normaliseDataset` which loops over the keys that are present in the `DOMStringMap` which I believe will always be strings.

Arguably it could be a private method that we don't export, or we integrate it into the `normaliseDataset` method, but that does make it slightly messier to test.

Reply:
Yeah that it really, if it's "public" code and accepts any `DOMStringMap` value I'd add the checks

Otherwise all good as it is, happy to get this merged